### PR TITLE
#14 Create the Button component (Style Guide Components)

### PR DESCRIPTION
**Pull request by BolajiOlajide**
#### What does this PR do
Create the Button component (Style Guide Components)

#### Description of Task to be completed
* Configure and Install storybook for this project
* Add a PULL REQUEST template for contributions
* Add `.editorconfig` for automatic linting.
* Create a reusable button component according to the [style guide](https://www.figma.com/file/wtojDaoY2PJED946ZMIvsX/OSCA-style-guide%2Fpattern?node-id=0%3A1).

#### How should this be manually tested
By opening up the storybook and viewing the button component.
The storybook can be stated with the command `yarn storybook` or `npm run storybook`

#### What are the relevant issues this PR belongs to
[#14](https://github.com/oscommunityafrica/osf-website/issues/14)

#### Any background context you want to add
N/A

#### Screenshots
Screenshot 2019-11-28 at 3 28 47 PM

### CODE REVIEW

File: src/pages/about.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { graphql, useStaticQuery} from "gatsby";
+import { graphql, useStaticQuery } from "gatsby";
```
Comment (BolajiAyodeji) [inline src/pages/about.js]: Thanks for this fix

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#### What does this PR do
```
Comment (BolajiAyodeji) [inline .github/PULL_REQUEST_TEMPLATE.md]: Thanks for this :+1:

**Comment by BolajiOlajide** [comment]
Here's a question I should have asked before starting this ticket: What is the default color and background for this Button component?

From the Figma link, there are about 3 variations and I'm not sure which should be the default but I made the blue the default.  @BolajiAyodeji

File: src/components/button/style.module.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.btn_container {
+  border-radius: 6px;
+  background: #283756;
+  padding: 10px;
+  color: #FFF;
+  min-width: 150px;
+  cursor: pointer;
+}
+
```
Comment (BolajiAyodeji) [inline src/components/button/style.module.css]: The focus state has a blue border around it onClick. Might need this?

```css
.btn_container:focus {
  outline: 0;
}
```

**Comment by BolajiAyodeji** [comment]
> Here's a question I should have asked before starting this ticket: What is the default color and background for this Button component?
> 
> From the Figma link, there are about 3 variations and I'm not sure which should be the default but I made the blue the default. @BolajiAyodeji

The default color is `#283756` and the design is clearly labeled or maybe you have the wrong file?

There are seven default color variations of the button with four states (Default, Hover, Active and Disabled)
Here's a snapshot of the default states:

image

File: src/components/button/style.module.css (hunk 3 above)
Comment (BolajiOlajide) [reply src/components/button/style.module.css]: I actually didn't add that intentionally, the focus ring is used to signify focus for users that access the application without a pointer device (e.g mouse).

You can have a look at [this](https://a11yproject.com/posts/never-remove-css-outlines/) and [this](https://medium.com/hackernoon/removing-that-ugly-focus-ring-and-keeping-it-too-6c8727fefcd2) for alternatives but I'll advise against removing the outline as it serves a purpose (especially considering we want to make this accessible for everyone).

That said if the decision is to remove it, I'm fine with removing the outline. Just say the word.

**Comment by BolajiOlajide** [comment]
> > Here's a question I should have asked before starting this ticket: What is the default color and background for this Button component?
> > From the Figma link, there are about 3 variations and I'm not sure which should be the default but I made the blue the default. @BolajiAyodeji
> 
> The default color is `#283756` and the design is clearly labeled or maybe you have the wrong file?
> 
> There are seven default color variations of the button with four states (Default, Hover, Active and Disabled)
> Here's a snapshot of the default states:
> 
> image

Okay, so it's my mistake - I should have asked before picking this ticket; hence, my initial apologies for asking late.
The idea of something being default means it's the fallback in case something doesn't exist - so you can't say you have `seven default color variations of the button` because a default implies a single entity.

Anyways, for the variations requested, there's going to be some sort of trigger to choose which variation to use. What name do you think we name this property and what should the different variations be called?

We can create an `enum`-like object called variations that can be imported by anyone? What should we then call the individual variation?

File: src/components/button/style.module.css (hunk 3 above)
Comment (BolajiAyodeji) [reply src/components/button/style.module.css]: Thanks for the clarification, we should leave it as we want this to be accessible.

**Comment by BolajiAyodeji** [comment]
> > > Here's a question I should have asked before starting this ticket: What is the default color and background for this Button component?
> > > From the Figma link, there are about 3 variations and I'm not sure which should be the default but I made the blue the default. @BolajiAyodeji
> > 
> > 
> > The default color is `#283756` and the design is clearly labeled or maybe you have the wrong file?
> > There are seven default color variations of the button with four states (Default, Hover, Active and Disabled)
> > Here's a snapshot of the default states:
> > image
> 
> Okay, so it's my mistake - I should have asked before picking this ticket; hence, my initial apologies for asking late.
> The idea of something being default means it's the fallback in case something doesn't exist - so you can't say you have `seven default color variations of the button` because a default implies a single entity.
> 
> Anyways, for the variations requested, there's going to be some sort of trigger to choose which variation to use. What name do you think we name this property and what should the different variations be called?
> 
> We can create an `enum`-like object called variations that can be imported by anyone? What should we then call the individual variation?

Sorry for the delayed response, I had some power outage.
I just clarified from the design team, for this design we have 7 variations of buttons and each has a default state but for each variation. Apologies for the confusion.

We're already using the `oc` prefix for classes, we can also apply this for the buttons. For simplicity, let's use numbers for the individual variation like so:

(In ascending order according to the design)

oc_btn_01
oc_btn_02
oc_btn_03
oc_btn_04
oc_btn_05
oc_btn_06
oc_btn_07

What do you think?

**Comment by BolajiOlajide** [comment]
@BolajiAyodeji Sounds good. Something the designer needs to consider is the contrast for the disabled buttons, they'll be displayed poorly for folks with poor vision - the white color on a very faint shade of blue might be hard for users to see even though the button is disabled. I think it's something to consider sha.

I'll be pushing up something soon, that follows the exact colors and details as shown in the style guide.

**Comment by BolajiAyodeji** [comment]
> @BolajiAyodeji Sounds good. Something the designer needs to consider is the contrast for the disabled buttons, they'll be displayed poorly for folks with poor vision - the white color on a very faint shade of blue might be hard for users to see even though the button is disabled. I think it's something to consider sha.
> 
> I'll be pushing up something soon, that follows the exact colors and details as shown in the style guide

I'll cc @perriefidelis (the design lead) on this, hopefully she can make changes.
Thanks for the great work chief.

**Comment by BolajiOlajide** [comment]
Alright.
As I shared it's something to consider, not trying to derail the design direction.

I've implemented the feedback shared on this PR anyways.

**Comment by BolajiAyodeji** [comment]
> Alright.
> As I shared it's something to consider, not trying to derail the design direction.
> 
> I've implemented the feedback shared on this PR anyways.

Thanks for this chief, can you check the conflicts, please?

**Comment by BolajiOlajide** [comment]
Done, sir.

**Comment by BolajiAyodeji** [comment]
Thanks for this PR chief!

**Comment by BolajiAyodeji** [comment]
@allcontributors please add @BolajiOlajide for code,doc

**Comment by BolajiAyodeji** [comment]
@allcontributors please add @BolajiOlajide for code and doc

**Comment by perriefidelis** [comment]
> @BolajiAyodeji Sounds good. Something the designer needs to consider is the contrast for the disabled buttons, they'll be displayed poorly for folks with poor vision - the white color on a very faint shade of blue might be hard for users to see even though the button is disabled. I think it's something to consider sha.
> 
> I'll be pushing up something soon, that follows the exact colors and details as shown in the style guide.

the disabled button will mostly not be used, not on the website obviously ... the style guide was not just for the website it constitutes of a guide for all design pushed from osca(flyers, web, swags etc) ... sorry for all the confusion guys, and my nonavailability in the past few days ... reach out me if there's any more confusion on this

**Pull request merged**